misc/cf-support

```diff
@@ -0,0 +1,165 @@
+.br
+Binary packages may be downloaded from http://cfengine.com/downloads/.
+.br
+The source code is available at http://github.com/cfengine/
+.SH BUGS
+Please see the public bug-tracker at http://bug.cfengine.com/.
+.br
+GitHub pull-requests may be submitted to http://github.com/cfengine/core/.
+.SH "SEE ALSO"
+.BR cf-promises (8),
+.BR cf-agent (8),
+.BR cf-serverd (8),
+.BR cf-execd (8),
+.BR cf-monitord (8),
+.BR cf-runagent (8),
+.BR cf-key (8)
+.SH AUTHOR
+Northern.tech AS
+EOF
+exit 0
+fi
+
+[ "$1" = "--yes" ] || [ "$1" = "-y" ] && non_interactive=1
+set -e
+if [ "$EUID" -ne 0 ]; then
+  echo "$0 must be run as root"
+  exit 1
+fi
+
+
+if [ $non_interactive -eq 0 ]; then
+  read -rp "If you have it, please enter your support case number: " case_number
+fi
+case_number=${case_number:-NA}
+timestamp=$(date +%F-%H%M)
+collection="cfengine_support_case_$case_number-$timestamp"
+
+tmpdir="$(mktemp -d)/$collection"
+export tmpdir
+mkdir -p "$tmpdir"
+
+export info_file="$tmpdir/system-info.txt"
+
+function file_add
+{
+  local filename
+  filename="$(basename "$1")"
+  if [ -f "$1" ]; then
+    cp "$1" "$tmpdir"/"$filename"
+  else
+    echo "$1 not found" >> "$info_file"
+  fi
+}
+export -f file_add
+
+function gzip_add
+{
+  local filename
+  filename="$(basename "$1")"
+  if [ -f "$1" ]; then
+    gzip -c "$1" > "$tmpdir"/"$filename".gz
+  else
+    echo "$1 not found" >> "$info_file"
+  fi
+}
+export -f gzip_add
+
+function log_cmd
+{
+  echo "=== $1 ===" >> "$info_file"
+  eval "$1" >> "$info_file" 2>&1 || true
+  echo >> "$info_file" # one newline for easier to read output
+}
+export -f log_cmd
+
+log_cmd "hostname -f"
+log_cmd "uname -a"
+[ -f "/etc/os-release" ] && log_cmd "cat /etc/os-release"
+log_cmd "$BINDIR/cf-promises -V"
+log_cmd "free -m"
+log_cmd "df -h"
+log_cmd "ps auwwx"
+log_cmd "top -b -H -c -n1"
+log_cmd "netstat -ie"
+command -v lsof 2>/dev/null >/dev/null && lsof > "$tmpdir/lsof.txt" || true
+log_cmd "mount"
+file_add "/etc/fstab"
+log_cmd "$BINDIR/cf-key -p $WORKDIR/ppkeys/localhost.pub"
+log_cmd "grep 'version =' $WORKDIR/inputs/promises.cf"
+log_cmd "$BINDIR/cf-key -s"
+log_cmd "$BINDIR/cf-check diagnose"
+if [ "$(command -v systemctl >/dev/null)" ]; then
+  log_cmd "systemctl status cfengine3"
+else
+  log_cmd "/etc/init.d/cfengine3 status"
+fi
+for f in /var/log/CFEngine-Install*; do
+  gzip_add "$f"
+done
+[ -f /var/log/messages ] && syslog_cmd="cat /var/log/messages"
+[ -f /var/log/syslog ] && syslog_cmd="cat /var/log/syslog"
+[ -z "$syslog_cmd" ] && syslog_cmd="dmesg"
+_syslog_filtered="$tmpdir"/syslog-filtered-for-cfengine.log.gz
+$syslog_cmd | grep -E 'cf-|CFEngine' | gzip -c > "$_syslog_filtered" || true
+gzip_add $WORKDIR/outputs/previous
+file_add $WORKDIR/policy_server.dat
+
+if [ -f $WORKDIR/share/cf-support-nova-hub.sh ]; then
+  # shellcheck source=/dev/null
+  source $WORKDIR/share/cf-support-nova-hub.sh
+fi
+
+tar czf "$collection.tar.gz" --directory "$tmpdir/../" . && rm -rf "$tmpdir"
```

Comment:
Why `/../`?

Reply:
`tmpdir="$(mktemp -d)/$collection"` so we use `mktemp` and add another dir `$collection` and want the support tarball to have that top-level directory `$collection` in the tarball.

Reply:
OK, makes sense. Please add a comment explaining this.